lib/course_api.rb

```diff
@@ -0,0 +1,153 @@
+require 'terms'
+require 'json'
+require 'nokogiri'
+require 'faraday'
+
+# Get course term information and then parse
+class CourseAPI
+    def initialize(xml)
+        @course_term = Nokogiri::XML(xml)
+    end
+    # Parse the XML returned by the CourseTerm MaIS API
+    def parse
+        courses = []
+        all_errors = []
+        # Return array of course hash objects
+        courses_list = parse_courses()
+        counter = 0
+        courses_list.each do |course|
+            counter += 1
+            if(counter == 10000)
+                sleep(1)
+                counter = 0
+            end 
+            request_class_id = course[:request_class_id]
+            course_info = get_course_info(request_class_id)
+            sections = course_info[:sections]
+            all_errors.concat(course_info[:errors]) 
+            sections.each do |section|
+                section_id = section[:sid]
+                instructors = section[:instructors]
+                course_hash = {"title": course[:title], "term": course[:term], "cid": course[:cid], "cids": course[:cids], 
```

Comment:
I'm assuming string-type keys are necessary for compatibility with the old behavior or database serialization, or could we use `with_indifferent_access` and 🙈 ? The mix of string and symbol keys makes it a little hard to follow.